# SAPM Assignment 1

Download the daily last 6 month closing prices for the common stock of your assigned company and calculate daily returns, expected daily return (assuming equal probability for observed historical data) and risk (standard deviation) for this expected return

In [ ]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
sns.set()

In [4]:
raw = pd.read_csv('501904063_Dabur.csv')

raw.head()


Symbol Series        Date  Prev Close  Close Price  Unnamed: 5  Unnamed: 6  \
0  DABUR     EQ  17-01-2020      487.60       489.65         NaN         NaN   
1  DABUR     EQ  20-01-2020      489.65       486.05         NaN         NaN   
2  DABUR     EQ  21-01-2020      486.05       487.00         NaN         NaN   
3  DABUR     EQ  22-01-2020      487.00       486.80         NaN         NaN   
4  DABUR     EQ  23-01-2020      486.80       489.80         NaN         NaN   

   Unnamed: 7  Unnamed: 8  
0         NaN         NaN  
1         NaN         NaN  
2         NaN         NaN  
3         NaN         NaN  
4         NaN         NaN

In [6]:
dropp = ['Unnamed: 5','Unnamed: 6','Unnamed: 7','Unnamed: 8']
data = raw.drop(dropp,axis =1)
data.head()

Symbol Series        Date  Prev Close  Close Price
0  DABUR     EQ  17-01-2020      487.60       489.65
1  DABUR     EQ  20-01-2020      489.65       486.05
2  DABUR     EQ  21-01-2020      486.05       487.00
3  DABUR     EQ  22-01-2020      487.00       486.80
4  DABUR     EQ  23-01-2020      486.80       489.80

In [7]:
data['Return'] = (data['Close Price'] - data['Prev Close'])/data['Prev Close']

data.head()

Symbol Series        Date  Prev Close  Close Price    Return
0  DABUR     EQ  17-01-2020      487.60       489.65  0.004204
1  DABUR     EQ  20-01-2020      489.65       486.05 -0.007352
2  DABUR     EQ  21-01-2020      486.05       487.00  0.001955
3  DABUR     EQ  22-01-2020      487.00       486.80 -0.000411
4  DABUR     EQ  23-01-2020      486.80       489.80  0.006163

In [8]:
data.describe()

Prev Close  Close Price      Return
count  124.000000   124.000000  124.000000
mean   470.942339   470.967742    0.000351
std     28.692713    28.708971    0.024582
min    396.050000   396.050000   -0.108698
25%    448.475000   448.475000   -0.010226
50%    473.150000   473.150000    0.001076
75%    495.837500   495.837500    0.009267
max    519.900000   519.900000    0.116293

In [9]:
data['Probability'] = 1/124

data.describe()

Prev Close  Close Price      Return   Probability
count  124.000000   124.000000  124.000000  1.240000e+02
mean   470.942339   470.967742    0.000351  8.064516e-03
std     28.692713    28.708971    0.024582  5.225283e-18
min    396.050000   396.050000   -0.108698  8.064516e-03
25%    448.475000   448.475000   -0.010226  8.064516e-03
50%    473.150000   473.150000    0.001076  8.064516e-03
75%    495.837500   495.837500    0.009267  8.064516e-03
max    519.900000   519.900000    0.116293  8.064516e-03

In [25]:
E_Ret = np.dot(data['Return'],data['Probability'])

E_Ret_P = "{:.5%}".format(E_Ret)

avg_ret = "{:.5%}".format(data['Return'].mean())

std_ret = "{:.5%}".format(data['Return'].std())



print("Expected Return assuming equal probability of the 124 observed historical daily returns:", E_Ret_P,"\n\n"
     ,"Average Return:", avg_ret,"\n\n"
     ,'Standard Deviation of this daily return:', std_ret,"\n\n"
     ,'Coefficient of Variation:', data['Return'].std()/data['Return'].mean())

Expected Return assuming equal probability of the 124 observed historical daily returns: 0.03513% 

 Average Return: 0.03513% 

 Standard Deviation of this daily return: 2.45823% 

 Coefficient of Variation: 69.9762999769152


Expected Return assuming equal probability of the 124 observed historical daily returns:  0.03513% 

Average Return:  0.03513% 

Standard Deviation of this daily return:  2.45823% 

Coefficient of Variation:  69.9762999769152

Risk = Standard Deviation

The coefficient of variation (CV) is a statistical measure of the relative dispersion of data points in a data series around the mean. In finance, the coefficient of variation allows investors to determine how much volatility, or risk, is assumed in comparison to the amount of return expected from investments